tests/integration/test_users.py

```diff
@@ -200,7 +201,13 @@ def test_patch_user(client, auth_headers, new_user):
     assert changeOwnRoleResponse.json == \
             {'message': 'Only admins can change roles'}
 
-
+def test_unique_user_constraint(client, auth_headers, new_user):
+    """Emails must be unique, otherwise an Exception is thrown"""
+    with pytest.raises(Exception) as e_info:
+        userToPatch = UserModel.find_by_email(new_user.email)
+        response = client.patch(f"/api/user/{userToPatch.id}",
+                           json = {"email": "user1@dwellingly.org"},
+                           headers=auth_headers["admin"])
```

Comment:
Writing validation tests will hopefully be a lot simpler with marshmallow. Usually I wouldn't test this type of database configuration, because the constraint would be repeated on the application side. This test is good in that it highlights we will send an uncaught 500 internal server error to the frontend. Do you think we should add the application side validation check now to mitigate this or should we wait and do it with the future refactorings?

Reply:
If the uncaught exception is going to cause problems for the FE development, then we should add the application side validation check now. For example, if the FE expects a failure status code, or a failure message in cases where the email is a duplicate, then leaving the exception will be a problem. Otherwise, I think we can wait until we refactor.

Reply:
I don't think the FE is handling that scenario. It can probably wait without any issues.